src/main/sarl/fr/utbm/info/vi51/worldswar/environment/Environment.sarl

```diff
@@ -79,17 +79,20 @@ agent Environment {
 			this.waitingAgentsInitialization = false
 			
 			//TODO for testing purpose, should be removed when we have ants spawners
-			//for(var i=0; i<500; i++) {
-			//	spawnAnt(new Colony(Breed.DARK_ANTS), Caste.GATHERER, new Point(0,0))
-			//}
-			//info("Ants spawned")
+//			info("Spawning ants")
+//			for(var i=0; i<500; i++) {
+//				spawnAnt(new Colony(Breed.DARK_ANTS), Caste.GATHERER, new Point(0,0))
+//			}
+//			info("Ants spawned")
 			
 			emit(new EnvironmentInitializedEvent)
 		}
 	}
 	
 	on StopSimulationEvent {
 		synchronized(this) {
+			info("Environment stopping")
+			emit(new KillAgentEvent)
 			killMe
```

Comment:
On est sure que ça ne pose aucun problème de faire un killme dans un synchronized(this) ?